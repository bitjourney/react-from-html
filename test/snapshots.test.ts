import "tsconfig-paths/register";
import renderer from "react-test-renderer";
import { ReactFromHtml } from "./entrypoint";

describe("ReactFromHtml#parse", () => {
  const reactFromHtml = new ReactFromHtml();

  it("renders single block of HTML elements correctly", () => {
    const tree = renderer
      .create(
        reactFromHtml.parse(
          "<p class='hello'>Hello, <strong>React</strong> world!</p>"
        )
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders multiple blocks of HTML elements correctly", () => {
    const tree = renderer
      .create(
        reactFromHtml.parse(`
          <p>Hello, <strong>React</strong> world!</p>
          <p>Hello, <strong>React</strong> world!</p>
          <p>Hello, <strong>React</strong> world!</p>
        `)
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders <script/> with dangerouslySetInnerHTML", () => {
    const tree = renderer
      .create(
        reactFromHtml.parse(
          "<script>console.log('Hello <strong>React</strong> world!');</script>"
        )
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders <style/> with dangerouslySetInnerHTML", () => {
    const tree = renderer
      .create(
        reactFromHtml.parse(
          "<style>/* Hello <strong>React</strong> world! */</style>"
        )
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders <textarea/> with defaultValue, instead of children", () => {
    const tree = renderer
      .create(
        reactFromHtml.parse(
          "<textarea>/* Hello <strong>React</strong> world! */</textarea>"
        )
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders <input/> with defaultValue", () => {
    const tree = renderer
      .create(
        reactFromHtml.parse(
          "<input type='text' value='foo' size='10' disabled/>"
        )
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders for as htmlFor", () => {
    const tree = renderer
      .create(reactFromHtml.parse("<label for='foo'><input id='foo'/></label>"))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("does not render react-reserved attributes", () => {
    const tree = renderer
      .create(reactFromHtml.parse("<div innerHTML=\"<input id='foo'/>\"/>"))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders elements with the style attribute", () => {
    const tree = renderer
      .create(
        reactFromHtml.parse(
          "<label style='color: red; font-size: 20px; -webkit-font-smoothing: antialiased; opacity: 0.5; font-weight: normal; z-index: 99'>Hello, world!</div>"
        )
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
