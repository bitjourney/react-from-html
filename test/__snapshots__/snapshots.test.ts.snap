// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactFromHtml#parse renders <input/> with defaultValue 1`] = `
<input
  defaultValue="foo"
  type="text"
/>
`;

exports[`ReactFromHtml#parse renders <script/> with dangerouslySetInnerHTML 1`] = `
<script
  dangerouslySetInnerHTML={
    Object {
      "__html": "console.log('Hello <strong>React</strong> world!');",
    }
  }
/>
`;

exports[`ReactFromHtml#parse renders <style/> with dangerouslySetInnerHTML 1`] = `
<style
  dangerouslySetInnerHTML={
    Object {
      "__html": "/* Hello <strong>React</strong> world! */",
    }
  }
/>
`;

exports[`ReactFromHtml#parse renders <svg/> correctly 1`] = `
Array [
  "
          ",
  <svg
    height="60"
    version="1.1"
    width="60"
    xmlns="http://www.w3.org/2000/svg"
  >
    
            
    <rect
      height="60"
      width="60"
    />
    
            
    <text
      alignmentBaseline="central"
      style={Object {}}
      textAnchor="middle"
      x="30"
      y="30"
    >
      Hello, world!
    </text>
    
          
  </svg>,
  "
        ",
]
`;

exports[`ReactFromHtml#parse renders <textarea/> with defaultValue, instead of children 1`] = `
<textarea
  defaultValue="/* Hello <strong>React</strong> world! */"
/>
`;

exports[`ReactFromHtml#parse renders for as htmlFor 1`] = `
<label
  htmlFor="foo"
>
  <input
    id="foo"
  />
</label>
`;

exports[`ReactFromHtml#parse renders multiple blocks of HTML elements correctly 1`] = `
Array [
  "
          ",
  <p>
    Hello, 
    <strong>
      React
    </strong>
     world!
  </p>,
  "
          ",
  <p>
    Hello, 
    <strong>
      React
    </strong>
     world!
  </p>,
  "
          ",
  <p>
    Hello, 
    <strong>
      React
    </strong>
     world!
  </p>,
  "
        ",
]
`;

exports[`ReactFromHtml#parse renders single block of HTML elements correctly 1`] = `
<p
  className="hello"
>
  Hello, 
  <strong>
    React
  </strong>
   world!
</p>
`;
